### PR TITLE
Update docs about new internal compiler

### PR DESCRIPTION
**Pull request by jhump**
This links to the new compiler instead of the old one. It also updates the benchmark number and revises the caveats to describe the new compiler.

I will hold off merging until https://github.com/bufbuild/buf/pull/1463/ is merged and a new release is made.

### CODE REVIEW

File: docs/reference/internal-compiler.md [hunk 1]
```diff
@@ -62,24 +62,27 @@ the byte representation of a serialized `FileDescriptorSet` produced by `buf`
...
+(which `buf` handles in all cases), `buf`'s compiler is _faster_ than
 `protoc` in most scenarios. `buf` parses your `.proto` files across all
 available cores, and re-orders the result to match `protoc`'s ordering as a
-post-processing task. As an example, `buf` can compile all 2,311 `.proto` files
-in [googleapis](https://github.com/googleapis/googleapis) in about 0.8s, on a
-four-core machine, as opposed to about 4.3s for `protoc` on the same machine.
+post-processing task. As an example, `buf` can compile all 3,944 `.proto` files
+in [googleapis](https://github.com/googleapis/googleapis/tree/cb6fbe8784479b22af38c09a5039d8983e894566)
+in about 0.9s, on a 2022 Macbook Pro with an M1 Max chip, as opposed to about
+1.6s for `protoc` v21.7 on the same machine.
```
Comment (jhump) [inline docs/reference/internal-compiler.md:85]: protoc seems to have accelerated quite a bit. This is either due to on-going performance improvements they are making, or perhaps their C++ compiler output just runs better/more optimally on an M1 chip than the Go compiler output. In any event, buf still has an advantage (just no longer "considerable").

File: docs/reference/internal-compiler.md [hunk 2]
```diff
@@ -62,24 +62,27 @@ the byte representation of a serialized `FileDescriptorSet` produced by `buf`
 and by `protoc`, and they are equal. There are two known exceptions that make
 this not always the case:
 
-1. `buf` produces additional intermediate SourceCodeInfo, and retains more
-   detached comments, than `protoc`. This is strictly more information for
-   consumers of the resulting `FileDescriptorSet`s.
+1. `buf` produces slightly different SourceCodeInfo for the special `json_name`
```
Comment (mfridman) [inline docs/reference/internal-compiler.md:65]: Without digging into the internals, is it "slightly different" or is it "additional"? 

The original sentence made it seem `buf` was on par with `protoc` and then some. Whereas this now reads there are notable differences.

**Review by mfridman — APPROVED** [review]
Overall looks good, just one question.

File: docs/reference/internal-compiler.md [hunk 3]
```diff
@@ -48,7 +48,7 @@ that aims to manage your Protobuf schema. Therefore, we've taken a different
 route.
 
 The
-[internal compiler](https://godoc.org/github.com/jhump/protoreflect/desc/protoparse)
+[internal compiler](https://godoc.org/github.com/bufbuild/protocompile)
```
Comment (jhump) [inline docs/reference/internal-compiler.md:51]:
```suggestion
[internal compiler](https://pkg.go.dev/github.com/bufbuild/protocompile)
```

**Review by pkwarren — APPROVED** [review]

**Pull request merged**